internal/uvm/vsmb.go

```diff
@@ -234,21 +238,36 @@ func (uvm *UtilityVM) AddVSMB(ctx context.Context, hostPath string, options *hcs
 // RemoveVSMB removes a VSMB share from a utility VM. Each VSMB share is ref-counted
 // and only actually removed when the ref-count drops to zero.
 func (uvm *UtilityVM) RemoveVSMB(ctx context.Context, hostPath string, readOnly bool) error {
+	st, err := os.Stat(hostPath)
+	if err != nil {
+		return err
+	}
+	isDir := st.IsDir()
+	if !isDir {
+		hostPath = filepath.Dir(hostPath)
+	}
```

Comment:
why do we need to do this if we've already determined that a share is or isn't a directory in AddVSMB?

Reply:
Mostly backwards compat, but also, `RemoveVSMB` may be called on the original file name (instead of via `Release`), so we still need the logic